api/v1alpha1/nodepool_types.go

```diff
@@ -55,62 +55,86 @@ type NodePool struct {
 	// Spec is the desired behavior of the NodePool.
 	Spec NodePoolSpec `json:"spec,omitempty"`
 
-	// Status is the most recently observed status of the NodePool.
+	// Status is the latest observed status of the NodePool.
 	Status NodePoolStatus `json:"status,omitempty"`
 }
 
-// NodePoolSpec defines the desired state of NodePool
+// NodePoolSpec is the desired behavior of a NodePool.
 type NodePoolSpec struct {
-	// ClusterName is the name of the Cluster this object belongs to.
+	// ClusterName is the name of the HostedCluster this NodePool belongs to.
+	//
+	// TODO(dan): Should this be a LocalObjectReference?
 	// +immutable
 	ClusterName string `json:"clusterName"`
+
+	// Release specifies the OCP release used for the NodePool. This informs the
+	// ignition configuration for machines, as well as other platform specific
+	// machine properties (e.g. an AMI on the AWS platform).
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Release Release `json:"release"`
+
+	// Platform specifies the underlying infrastructure provider for the NodePool
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform NodePoolPlatform `json:"platform"`
+
+	// NodeCount is the desired number of nodes the pool should maintain. The
+	// default value is 1.
```

Comment:
>default value is 1.

Actually we have no default here and that's fine. No NodeCount/0 is a valid use case and we also want to differentiate between 0 and nil for autoscaling.

Reply:
Thanks, fixed